pixeltable/utils/__init__.py

```diff
@@ -35,3 +37,22 @@ def sha256sum(path: Union[Path, str]) -> str:
             h.update(chunk)
 
     return h.hexdigest()
+
+
+def parse_file_or_url(file_or_url: str) -> Union[Path, urllib.parse.ParseResult]:
```

Comment:
How about just parse_local_path() or something like that and return an Optional[Path]? Unions are harder to follow and calling urlparse() again isn't a big deal.

Reply:
Ok yeah this is better. Done.